src/parser/SparqlParser.cpp

```diff
@@ -89,8 +89,8 @@ void SparqlParser::parseQuery(ParsedQuery* query, QueryType queryType) {
       auto& constructClause = query->constructClause();
       for (const auto& triple : constructClause) {
         for (const auto& varOrTerm : triple) {
-          if (std::holds_alternative<Variable>(varOrTerm)) {
-            const auto var = varOrTerm.toString();
+          if (auto variable = std::get_if<Variable>(&varOrTerm)) {
+            const auto var = variable->getName();
```

Comment:
`const auto&` (it is safe in general, because const references extend lifetime of returned temporaries, so it would even be ok, if `getName()` returned by value). And in this case,
I think you safe a copy, because `getName()` returns a reference.